### ISSUE TITLE
Type Error during incremental compilation 

### ISSUE THREAD

**Issue by stefanocke** [stefanocke/angular-cli-app-with-rollup#3]
Probably introduced with Angular 8.1.2 ?

```
TypeError: Cannot read property 'has' of null
    at C:\Java\github\angular-cli-app-with-rollup\node_modules\@angular\compiler-cli\src\ngtsc\incremental\src\state.js:120:84
    at Array.some (<anonymous>)
    at IncrementalState.hasChangedResourceDependencies (C:\Java\github\angular-cli-app-with-rollup\node_modules\@angular\compiler-cli\src\ngtsc\incremental\src\state.js:120:18)
    at IncrementalState.safeToSkip (C:\Java\github\angular-cli-app-with-rollup\node_modules\@angular\compiler-cli\src\ngtsc\incremental\src\state.js:70:57)
    at IvyCompilation.analyze (C:\Java\github\angular-cli-app-with-rollup\node_modules\@angular\compiler-cli\src\ngtsc\transform\src\compilation.js:156:39)
    at IvyCompilation.analyzeSync (C:\Java\github\angular-cli-app-with-rollup\node_modules\@angular\compiler-cli\src\ngtsc\transform\src\compilation.js:71:76)
    at C:\Java\github\angular-cli-app-with-rollup\node_modules\@angular\compiler-cli\src\ngtsc\program.js:257:39
    at Array.forEach (<anonymous>)
    at NgtscProgram.ensureAnalyzed (C:\Java\github\angular-cli-app-with-rollup\node_modules\@angular\compiler-cli\src\ngtsc\program.js:255:22)
    at NgtscProgram.getNgSemanticDiagnostics (C:\Java\github\angular-cli-app-with-rollup\node_modules\@angular\compiler-cli\src\ngtsc\program.js
```

**Comment by stefanocke** [comment]
see https://github.com/angular/angular/pull/31322

**Comment by stefanocke** [comment]
Fix seems not to be included in 8.1.3

**Comment by stefanocke** [comment]
8.2.0-rc.0 fixes the original bug but there are two (new?) issues now:

- watch mode ignores changes in html templates
- the position of the dynamic component has changed. it is the last one now
